components/datadog/agentparams/params.go

```diff
@@ -117,7 +120,19 @@ func WithAgentConfig(config string) func(*Params) error {
 // WithIntegration adds the configuration for an integration.
 func WithIntegration(folderName string, content string) func(*Params) error {
 	return func(p *Params) error {
-		p.Integrations[folderName] = content
+		confPath := path.Join("conf.d", folderName, "conf.yaml")
+		p.Files[confPath] = content
+		return nil
+	}
+}
+
+// WithFile adds a file to the install with the contents at the given path
+func WithFile(filePath string, content string) func(*Params) error {
```

Comment:
What do you think about reflecting that `filePath` is absolute in its naming (like `absolutePath`)?

Reply:
Fixed in https://github.com/DataDog/test-infra-definitions/pull/289/commits/225fcde25381273527fca1578144f561a0c25775